### PR TITLE
Implement Vec3.slerp to support spherical interpolation between directional vectors

### PR DESCRIPTION
**Pull request by shrinktofit**
Re: #

### Changelog

* Implement `Vec3.slerp` to support spherical interpolation between directional vectors.

See:

- Slerp on wiki: https://en.wikipedia.org/wiki/Slerp

- Unity's API: https://docs.unity3d.com/ScriptReference/Vector3.Slerp.html

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [x] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/core/math/vec3.ts [hunk 1]
```diff
@@ -1057,4 +1129,55 @@ export function v3 (x?: number | Vec3, y?: number, z?: number) {
     return new Vec3(x as any, y, z);
 }
 
+/**
+ * Chooses an arbitrary unit vector that is perpendicular to input.
+ */
+const chooseAnyPerpendicular = (() => {
```
Comment (stanleyljl) [inline cocos/core/math/vec3.ts]: It is better to drop the minimal absolute component in function chooseAnyPerpendicular

File: cocos/core/math/vec3.ts [hunk 2]
```diff
@@ -355,6 +356,77 @@ export class Vec3 extends ValueType {
...
+     * - Otherwise, if `from` and `to` have almost opposite directions,
+     * the result would be such a vector so that:
+     * The angle ratio between result vector and input vectors is `t`,
+     * the length of result vector is the linear interpolation of lengths from `from` to `to`.
+     *
+     * - Otherwise, the result would be the spherical linear interpolation result
+     * from normalized `from` to normalized `to`,
+     * then scaled by linear interpolation of lengths from `from` to `to`.
+     */
+    public static slerp= (() => {
```
Comment (stanleyljl) [inline cocos/core/math/vec3.ts:389]: Do we need to account for spherical interpolation of non-unit vectors?

Comment (shrinktofit) [reply cocos/core/math/vec3.ts:389]: This can implement, for example:

image

File: cocos/core/math/vec3.ts [hunk 3]
```diff
@@ -1132,31 +1132,22 @@ export function v3 (x?: number | Vec3, y?: number, z?: number) {
...
-            cacheV.y = v.z;
-            cacheV.z = -v.y;
-            cacheV.x = 0.0;
-        }
-        Vec3.normalize(out, cacheV);
-        return out;
-    };
-})();
+function chooseAnyPerpendicular (out: Vec3, v: Readonly<Vec3>) {
+    // 1. Drop the component with minimal magnitude.
```
Comment (shrinktofit) [inline cocos/core/math/vec3.ts:1137]: Suggestion from @stanleyljl . Thank you!

**Review by stanleyljl — APPROVED** [review]

**Pull request merged**